src/lib.rs

```diff
@@ -261,9 +261,11 @@ impl<'a> Uri<&'a str> {
 impl<'i, 'o, T: Io<'i, 'o> + AsRef<str>> Uri<T> {
     #[inline]
     /// Returns the URI reference as a string slice.
-    pub fn as_str(&'i self) -> &'o str {
-        // SAFETY: The indexes are within bounds and the validation is done.
-        unsafe { self.slice(0, self.len()) }
+    pub fn as_str(&self) -> &str {
+        // SAFETY: The indexes are within bounds.
+        let bytes = unsafe { slice::from_raw_parts(self.ptr.get(), self.len() as usize) };
```

Comment:
Coming in from https://github.com/gluon-lang/lsp-types/pull/266 .

This could really be a private `as_bytes()` method so the unsafe block isn't duplicated in PartialEq

Reply:
Good catch, fixed.